src/engine/renderer/tr_image.cpp

```diff
@@ -3049,6 +3049,13 @@ int numTextures = 0;
 
 qhandle_t RE_GenerateTexture( const byte *pic, int width, int height )
 {
+	size_t size = width * height;
+
+	ASSERT( !size );
+	if ( !size ) {
```

Comment:
Let's be consistent about style.

let's just do
```
if ( !size )
{
    return;
}
```

or

```
if ( !size ) return;
```

I think for single line if statements, we should avoid brackets.

Reply:
Done.

Reply:
If you write `ASSERT(size); if (!size) {...}`, with our ASSERT implementation, the compiler is allowed to optimize out the test. If you want to ensure the condition is always caught, better to replace the assert+if with a Log::Warn or Sys::Drop.

Reply:
The primary purpose of this patch is that a third-party individual cannot host a server providing an autodownloaded package that would either crash or abort the user's own game.

A `Log::Warn` would do the job if we want to report the error, but game execution should continue.

Reply:
Modified with a `Log::Warn`.

Reply:
> The primary purpose of this patch is that a third-party individual cannot host a server providing an autodownloaded package that would either crash or abort the user's own game.

If this is about catching mistakes made by the gamelogic, it should not use ASSERT at all. An ASSERT in the engine should be used when the engine detects violations of the assumptions of its own behavior.

Reply:
Then it's another good reason to use a `Log::Warn`.